frontend/server/src/DAO/Submissions.php

```diff
@@ -157,4 +157,86 @@ public static function countAcceptedSubmissions(
             [$startTimestamp, $endTimestamp]
         );
     }
+
+    /**
+     * @return list<array{time: int, username: string, school_id: int, school_name: string, alias: string, title: string, language: string, verdict: string, runtime: int, memory: int}>
+     */
+    public static function getLatestSubmissions(
+        int $page,
+        int $rowcount,
+        int $seconds = 3600 * 24
+    ): array {
+        $offset = ($page - 1) * $rowcount;
+        $sql = '
+            SELECT
+                UNIX_TIMESTAMP(s.time) as time,
+                i.username,
+                s.school_id,
+                sc.name as school_name,
+                p.alias,
+                p.title,
+                s.language,
+                r.verdict,
+                r.runtime,
+                r.memory,
+                COALESCE (
+                    (SELECT urc.classname
+                    FROM User_Rank_Cutoffs urc
+                    WHERE
+                        urc.score <= (
+                            SELECT
+                                ur.score
+                            FROM
+                                User_Rank ur
+                            WHERE
+                                ur.user_id = i.user_id
+                        )
+                    ORDER BY
+                        urc.percentile ASC
+                    LIMIT 1)
+                , "user-rank-unranked") AS classname
```

Comment:
```suggestion
                    (
                        SELECT
                            ...
                        LIMIT 1
                    ),
                    "user-rank-unranked"
                ) AS classname
```

en general las comas van en la línea de arriba y la línea siguiente alineada con el sangrado con el que empieza la línea anterior. y de paso puedes sangrar la subconsulta para que se vea más bonito, así también es consistente con las expresiones de L218-L224.

Reply:
Ok, creo que ahora sí quedó!